src/main/java/com/google/javascript/clutz/LegacyNamespaceReexportMapBuilder.java

```diff
@@ -0,0 +1,75 @@
+package com.google.javascript.clutz;
+
+import com.google.javascript.rhino.Node;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * When a goog.module imports a symbol, there's an ambiguity about whether the symbol is from a
+ * goog.provide style file or a goog.module style file. Normally this ambiguity is resolved by
+ * consulting depgraphs, but a goog.module can declareLegacyNamespace, in which case it's exports
+ * are in goog.provide style, but the depgraph says it's a goog.module. To resolve this ambiguity,
+ * LegacyNamespaceReexportMapBuilder generates mappings from the goog.module style symbols to the
+ * normally declared symbols, so DeclarationGenerator can produce aliases.
+ */
+public class LegacyNamespaceReexportMapBuilder extends ImportBasedMapBuilder {
+
+  @Override
+  protected Map<String, String> build(
+      String localModuleId, Node moduleBody, Set<String> googProvides) {
+    Map<String, String> reexportMap = new HashMap<>();
+    if (localModuleId == null) {
+      return reexportMap;
+    }
+
+    if (!isLegacyNamespaceModule(moduleBody)) {
+      return reexportMap;
+    }
+
+    for (Node statement : moduleBody.children()) {
+      if (isWholeModuleExportAssignment(statement)) {
```

Comment:
I feel like I have seen this code in AliasMapBuilder, is refactoring commonalities feasable?

Reply:
In general it would be nice to hide the `getFirstChild().getChildAtIndex(1)` type of AST mangling inside more meaningfully named reusable methods like what you did with isNamedExportAssignment.

Reply:
Done!